lab-cameron/lib/list.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const List = module.exports = function() {
+  for (let key in arguments) {
+    this[key] = arguments[key];
+  }
+  this.length = arguments.length;
+};
+
+List.prototype.copy = function(source) {
+  let result = source;
+  return result;
+};
+
+List.prototype.push = function(value) {
+  let result = this.copy(this);
+  result[result.length++] = value;
+  return result;
+};
+
+List.prototype.pop = function() {
+  let result = this.copy(this);
+  let val = result[result.length - 1];
+  delete result[--result.length];
+
+  return {
+    value: val,
+    list: result,
+  };
+};
+
+List.prototype.forEach = function(callback) {
+  for (let i = 0; i < this.length; i++) {
+    callback(this[i], i, this);
+  }
+};
+
+List.prototype.filter = function(value) {
```

Comment:
**TODO FOR LOST POINTS**

Your filter doesn't take a value, it takes a callback, then you want to modify your "if statement" at Line 41 to account for what to do if a callback is present. Slack me anytime even when I'm gone if you're still lost.